### PR TITLE
Update and fix environment and documentation

### PR DESCRIPTION
**Pull request by amotl**
Hi,

this is a mixture of things I still had on my working tree paired with some updates similar to https://github.com/crate/crate-pdo/pull/118 and https://github.com/crate/crate-pdo/pull/119.

- Update to crate-docs 2.0.0
- Run link checker each morning
- Update some outdated links
- Use Codecov instead of Coveralls
- Update Scrutinizer configuration
- Improve README and DEVELOP documents
- Remove notes about "enterprise edition" from documentation

With kind regards,
Andreas.

### CODE REVIEW

**Review by proddata — COMMENTED** [review]
too deep for me 😟

**Review by jayeff — APPROVED** [review]
only did a quick check but should be okay 😇

File: docs/appendices/compatibility.rst [hunk 1]
```diff
@@ -84,5 +79,4 @@ DBAL
 +----------------+----------------------------------------------+------------+
 
 .. _create your own users: https://crate.io/docs/crate/reference/en/latest/admin/user-management.html
```
Comment (seut) [inline docs/appendices/compatibility.rst:81]: unused? ;)

**Review by seut — APPROVED** [review]
👍 just maybe spotted another unused link ref

File: docs/appendices/compatibility.rst (hunk 1 above)
Comment (amotl) [reply docs/appendices/compatibility.rst]: Thank you for spotting this again.

**Pull request merged**